lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -114,32 +116,26 @@ static LogicalResult isPortDrivenByGroup(Value port, GroupOp groupOp) {
     }
   }
 
-  // If @p port is an output of an InstanceOp, and if any input port of
-  // this InstanceOp is driven within @p groupOp, we'll assume that @p port
-  // is sensitive to the driven input port.
-  // @TODO: simplify this logic when the calyx.cell interface allows us to more
-  // easily access the input and output ports of a component
-  if (auto instanceOp = dyn_cast<InstanceOp>(port.getDefiningOp())) {
-    auto compOp = instanceOp.getReferencedComponent();
-    auto compOpPortInfo = llvm::enumerate(getComponentPortInfo(compOp));
-
-    bool isOutputPort = llvm::any_of(compOpPortInfo, [&](auto portInfo) {
-      return port == instanceOp->getResult(portInfo.index()) &&
-             portInfo.value().direction == Direction::Output;
-    });
-
-    if (isOutputPort) {
-      return success(llvm::any_of(compOpPortInfo, [&](auto portInfo) {
-        return portInfo.value().direction == Direction::Input &&
-               succeeded(isPortDrivenByGroup(
-                   instanceOp->getResult(portInfo.index()), groupOp));
-      }));
+  // If @p port is an output of a cell then we conservatively enforce that all
+  // (and at least one) non-interface inputs of the cell must be driven by the
+  // group.
+  if (auto cell = dyn_cast<CellInterface>(port.getDefiningOp())) {
+    if (cell.direction(port) == calyx::Direction::Output) {
+      auto userDrivenPorts = cell.filterInterfacePorts(calyx::Direction::Input);
```

Comment:
Should the `userDrivenPorts.size() > 0` be an assertion? Seems like a programmer mistake to check if a given op is driven yet has no input ports.

Reply:
Will fix!